### PR TITLE
add state type endpoint

### LINKED ISSUE #13648: update state/get to display new format
Implement these 2 updates:
* [state/get](https://docs.google.com/document/d/1_Ru5_hzyr-NfqyL9T58RlkOMPLzeuKVl12UXJ2okOfw/edit#heading=h.yerfr86hiico)
* [state/type/get](https://docs.google.com/document/d/1_Ru5_hzyr-NfqyL9T58RlkOMPLzeuKVl12UXJ2okOfw/edit#heading=h.yaiqj1xpiesu)

### PR DESCRIPTION
**Pull request by cgardens**
closes https://github.com/airbytehq/airbyte/issues/13648

## What
* adds implementation for the state type endpoint `state/type/get`
* updates the implementation of `state/get` to use the new more rich state model
* fixes a bug where field transformation type enums couldn't convert between api and the internal data model

### CODE REVIEW

File: airbyte-server/src/main/java/io/airbyte/server/converters/StateConverter.java
Comment (benmoriceau) [inline airbyte-server/src/main/java/io/airbyte/server/converters/StateConverter.java:65]: Nit: we can use of instead of ofNullable

Comment (cgardens) [reply airbyte-server/src/main/java/io/airbyte/server/converters/StateConverter.java]: yup! fixed.

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java
Comment (jdpgrailsdev) [inline airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java:202]: Nit:  Consider constructing the `StatePersistence` object outside of this class and passing it in as an argument.  It will make the transition to dependency injection easier.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java
Comment (cgardens) [reply airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java]: legit. done.

**Review by benmoriceau — APPROVED** [review]

**Review by alovew — APPROVED** [review]

**Comment by cgardens** [comment]
This is blocked by a failing acceptance test. This is due to a bug found and described [here](https://github.com/airbytehq/airbyte/pull/14128#issuecomment-1166066974). @gosusnp is working on a patch. Once that is in, we can merge this.

**Pull request merged**
